taipy/core/data/couchbase_collection_datanode.py

```diff
@@ -0,0 +1,183 @@
+
+    The *properties* attribute can also contain the following optional entries:
+
+    - *db_username* (`str`): The username for the Couchbase database.
+    - *db_password* (`str`): The password for the Couchbase database.
+    - *db_host* (`str`): The database host. The default value is *"localhost"*.
+    - *db_port* (`int`): The database port. The default value is *8091*.
+    """
+
+    __STORAGE_TYPE = "couchbase_collection"
+
+    __DB_NAME_KEY = "db_name"
+    __COLLECTION_KEY = "collection_name"
+    __DB_USERNAME_KEY = "db_username"
+    __DB_PASSWORD_KEY = "db_password"
+    __DB_HOST_KEY = "db_host"
+    __DB_PORT_KEY = "db_port"
+
+    __DB_HOST_DEFAULT = "localhost"
+    __DB_PORT_DEFAULT = 8091
+
+    _REQUIRED_PROPERTIES: List[str] = [
+        __DB_NAME_KEY,
+        __COLLECTION_KEY,
+    ]
+
+    def __init__(
+        self,
+        config_id: str,
+        scope: Scope,
+        id: Optional[DataNodeId] = None,
+        owner_id: Optional[str] = None,
+        parent_ids: Optional[Set[str]] = None,
+        last_edit_date: Optional[datetime] = None,
+        edits: List[Edit] = None,
+        version: str = None,
+        validity_period: Optional[timedelta] = None,
+        edit_in_progress: bool = False,
+        editor_id: Optional[str] = None,
+        editor_expiration_date: Optional[datetime] = None,
+        properties: Dict = None,
+    ) -> None:
+        if properties is None:
+            properties = {}
+        required = self._REQUIRED_PROPERTIES
+        if missing := set(required) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties {', '.join(missing)} were not informed and are required."
+            )
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            owner_id,
+            parent_ids,
+            last_edit_date,
+            edits,
+            version or _VersionManagerFactory._build_manager()._get_latest_version(),
+            validity_period,
+            edit_in_progress,
+            editor_id,
+            editor_expiration_date,
+            **properties,
+        )
+
+
+         # Create a Couchbase connection using the provided properties.
+         # For more information on connecting to Couchbase, see:
+    # https://docs.couchbase.com/python-sdk/current/hello-world/start-using-sdk.html
+        try:
+            self.cluster = Cluster(
+                f"couchbase://{properties.get(self.__DB_HOST_KEY, self.__DB_HOST_DEFAULT)}",
+                ClusterOptions(
+                    PsswordAuthenticator(
+                        properties.get(self.__DB_USERNAME_KEY, ""),
+                        properties.get(self.__DB_PASSWORD_KEY, "")
+                    )
+                )
+            )
+            bucket = self.cluster.bucket(properties.get(self.__DB_NAME_KEY))
```

Comment:
```suggestion
            bucket = self.cluster.bucket(properties.get(self.__BUCKET_KEY))
```

The constant should be renamed to match what it actually presents